src/CommonDBTM.php

```diff
@@ -1851,9 +1851,9 @@ protected function manageLocks()
             && (!isset($this->input['is_dynamic']) || $this->input['is_dynamic'] == false)
         ) {
             $fields = array_values($this->updates);
-            $idx = array_search('date_mod', $fields);
-            if ($idx !== false) {
-                unset($fields[$idx]);
+            $idx = array_keys($fields, 'date_mod');
+            foreach ($idx as $i) {
+                unset($fields[$i]);
             }
```

Comment:
Using the `array_filter()` function seems easier to read.
```suggestion
            $fields = array_filter($fields, fn($f) => $f !== 'date_mod');
```